### PR TITLE
docs(node): Add instrux for restoring from snapshot

### PR DESCRIPTION
**Pull request by wbnns**
**What changed? Why?**

Add instructions to help people who want to restore from snapshot

**Notes to reviewers**

N/A

**How has it been tested?**

Localhost

**Does this PR add a new token to the bridge?**

Are you adding an entry to [`assets.ts`](../apps/bridge/assets.ts)?

- [X] No, this PR does not add a new token to the bridge
- [ ] Yes, and I've confirmed this token doesn't use a bridge override
- [ ] Yes, and I've confirmed this token is an OptimismMintableERC20

If you are adding a token to the bridge, please include evidence of both confirmations above for your reviewers.

### CODE REVIEW

File: apps/base-docs/tutorials/docs/0_run-a-base-node.md [hunk 1]
```diff
@@ -100,33 +100,47 @@ Syncing your node may take **days** and will consume a vast amount of your reque
...
-wget https://mainnet-full-snapshots.base.org/$(curl https://mainnet-full-snapshots.base.org/latest)
+wget https://sepolia-full-snapshots.base.org/$(curl https://sepolia-full-snapshots.base.org/latest)
 ```
 
-**Testnet (Sepolia)**
+**Testnet (Archive)**
 
 ```
-wget https://sepolia-full-snapshots.base.org/$(curl https://sepolia-full-snapshots.base.org/latest)
+wget https://base-sepolia-archive-snapshots.s3.us-east-1.amazonaws.com/$(curl https://base-sepolia-archive-snapshots.s3.us-east-1.amazonaws.com/latest)
```
Comment (zencephalon) [inline apps/base-docs/tutorials/docs/0_run-a-base-node.md:116]: I just tried this and it didn't work for me

File: apps/base-docs/tutorials/docs/0_run-a-base-node.md [hunk 2]
```diff
@@ -100,33 +100,47 @@ Syncing your node may take **days** and will consume a vast amount of your reque
...
+```
+wget https://mainnet-full-snapshots.base.org/$(curl https://mainnet-full-snapshots.base.org/latest)
+```
 
-**Mainnet**
+**Mainnet (Archive)**
 
 ```
-wget https://mainnet-archive-snapshots.base.org/$(curl https://mainnet-archive-snapshots.base.org/latest)
+wget https://base-mainnet-archive-snapshots.s3.us-east-1.amazonaws.com/$(curl https://base-mainnet-archive-snapshots.s3.us-east-1.amazonaws.com/latest)
```
Comment (zencephalon) [inline apps/base-docs/tutorials/docs/0_run-a-base-node.md:128]: This one also doesn't work for me

Comment (wbnns) [reply apps/base-docs/tutorials/docs/0_run-a-base-node.md]: @zencephalon 

Thanks, I had an outdated archive link; fixed. [47d0a7a](https://github.com/base-org/web/pull/457/commits/47d0a7a5a25d679aefa5c186acd069291cdcb15b)

File: apps/base-docs/tutorials/docs/0_run-a-base-node.md (hunk 1 above)
Comment (wbnns) [reply apps/base-docs/tutorials/docs/0_run-a-base-node.md:116]: Fixed: [47d0a7a](https://github.com/base-org/web/pull/457/commits/47d0a7a5a25d679aefa5c186acd069291cdcb15b)

**Review by zencephalon — DISMISSED** [review]

**Review by zencephalon — APPROVED** [review]

**Comment by jotto** [comment]
(moved this comment from https://github.com/base-org/node/pull/241)

@wbnns FWIW, the instructions you've put here didn't work for me (and apparently not for @ifsayyes in https://github.com/base-org/node/issues/239 either)

When I put the geth dir from the untar'd dir in my data dir, base/optimism's geth node did nothing for 24 hours... so I tried again by placing the snapshots dir (the root of the untar'd dir) in the geth data dir. Things started working, though it's still syncing (I predict 7 days total on huge fast machine), if it's taking that long, maybe the snapshot didn't take... but the first solution (the one you've got in this PR, definitely didn't work for me)

**Comment by wbnns** [comment]
@jotto Heya! Thanks for the quick feedback! Double-checking and will follow-up. 👍

**Comment by wbnns** [comment]
@jotto Hey, checking back in and just wanted to confirm that I was able to successfully restore from snapshot using these instructions.

1. Download snapshot and untar
2. Place the `geth` folder in the archive in your `geth-data` folder or other folder if the `DATA_DIR` was changed (make sure nothing else is in there)
3. Start node

2024-04-25 at 01 01 40@2x

The snapshot is about two days (2515 minutes) old.

**Comment by jotto** [comment]
Thank you for this confirmation... I don't know why my attempts (plural) failed, but I'll try again knowing someone else could make it work. I'll report back here.

**Comment by jotto** [comment]
Do you mind pasting some of the log output from the `base-node-geth` container (as opposed to `base-node-node`) - so I can see what log output for a successfully running geth instance reading from a snapshot looks like?

**Comment by jotto** [comment]
Here is the log output from my `base-node-geth` after dropping the geth dir from tar archive into data dir:

1 hour (and then no more) of "Generating state snapshot" every few seconds, final log entry for this looks like:

INFO [04-25|19:32:16.693] Generating state snapshot                root=614833..66c538 at=fef184..ed9b45 accounts=61,928,762 slots=372,846,964 storage=29.50GiB   dangling=0 elapsed=1h8m1.244s  eta=17.225s

and otherwise "Looking for peers" (it never finds any peers, I've waited up to 48 hours)

So maybe this is correct? Is this what a fully synced archive node looks like? I don't think so because If I query the RPC node for latest block I get `12234499` (seems to be stuck there, the snapshot I used was `base-mainnet-archive-1711261943.tar.gz` - March 24th.... which is also the date of 12234499)

```javascript
const latestBlockNumber = await web3.eth.getBlockNumber();
const block = await web3.eth.getBlock(latestBlockNumber, true);
console.log(`Block Number: ${block.number}`);
```

and I get "Latest synced block behind by: 46930 minutes" from the base RPC node... (32 days ago... date of the tar snapshot)

---

`base-node-node` logs look like:

t=2024-04-25T19:37:11+0000 lvl=info msg="Starting P2P sync client event loop" peer=16Uiu2HAkyzvaA1fnSftWSuwZf2qgPErNy4m9uei11mBYaDrww7Rh
t=2024-04-25T19:37:11+0000 lvl=info msg="attempting connection" peer=16Uiu2HAmG8YrXGL7AVbA5oetjagMr7zk4o1KGTrHii71fKpADjgZ
t=2024-04-25T19:37:11+0000 lvl=info msg="connected to peer" peer=16Uiu2HAmG8YrXGL7AVbA5oetjagMr7zk4o1KGTrHii71fKpADjgZ addr=/ip4/104.250.135.50/tcp/9222
t=2024-04-25T19:37:11+0000 lvl=info msg="Starting P2P sync client event loop" peer=16Uiu2HAmG8YrXGL7AVbA5oetjagMr7zk4o1KGTrHii71fKpADjgZ
t=2024-04-25T19:37:11+0000 lvl=info msg="attempting connection" peer=16Uiu2HAkvTfEA2Lrj2WchFL7egmd5nX6HGe6C8CjSBpkhMzCAq4u
t=2024-04-25T19:37:11+0000 lvl=info msg="Received signed execution payload from p2p" id=0xd75291e7499f3cd54e6c65d55d273f9296170252702d1fbd7079c3e4f5487284:13642242 peer=16Uiu2HAm3QKWSTr26LHAtsAxs4LwSQpvgGWkKFjGK6b2jnFdSFrC
t=2024-04-25T19:37:11+0000 lvl=info msg="Optimistically queueing unsafe L2 execution payload" id=0xd75291e7499f3cd54e6c65d55d273f9296170252702d1fbd7079c3e4f5487284:13642242
t=2024-04-25T19:37:12+0000 lvl=info msg="connected to peer" peer=16Uiu2HAkvTfEA2Lrj2WchFL7egmd5nX6HGe6C8CjSBpkhMzCAq4u addr=/ip4/65.108.224.241/tcp/9222
t=2024-04-25T19:37:12+0000 lvl=info msg="Starting P2P sync client event loop" peer=16Uiu2HAkvTfEA2Lrj2WchFL7egmd5nX6HGe6C8CjSBpkhMzCAq4u
t=2024-04-25T19:37:12+0000 lvl=info msg="attempting connection" peer=16Uiu2HAm4zgBaJF2xzkNpqLzQbdGgZUQAuuWrgBpaKQMyxSJdCAh
t=2024-04-25T19:37:13+0000 lvl=info msg="connected to peer" peer=16Uiu2HAm4zgBaJF2xzkNpqLzQbdGgZUQAuuWrgBpaKQMyxSJdCAh addr=/ip4/154.26.136.163/tcp/9222
t=2024-04-25T19:37:13+0000 lvl=info msg="Starting P2P sync client event loop" peer=16Uiu2HAm4zgBaJF2xzkNpqLzQbdGgZUQAuuWrgBpaKQMyxSJdCAh
t=2024-04-25T19:37:13+0000 lvl=warn msg="Derivation process temporary error" attempts=395 err="engine stage failed: temp: failed to fetch blobs: failed to get blob sidecars for L1BlockRef 0x214ad8ec43a9bc1c1280ce43c6ac55d4bc951b683c20cff2541e2933d0f3b510:19501581: expected 6 sidecars but got 0"
t=2024-04-25T19:37:13+0000 lvl=info msg="attempting connection" peer=16Uiu2HAmCzw9ezt1tHGeCqcFbGKrL5o6b6zmCtQDYoCywWZk4K6i
t=2024-04-25T19:37:13+0000 lvl=info msg="connected to peer" peer=16Uiu2HAmCzw9ezt1tHGeCqcFbGKrL5o6b6zmCtQDYoCywWZk4K6i addr=/ip4/51.81.242.36/tcp/9222
t=2024-04-25T19:37:13+0000 lvl=info msg="Starting P2P sync client event loop" peer=16Uiu2HAmCzw9ezt1tHGeCqcFbGKrL5o6b6zmCtQDYoCywWZk4K6i
t=2024-04-25T19:37:13+0000 lvl=info msg="attempting connection" peer=16Uiu2HAm8g9VXNqe5thUGoywAETRctjqqdeuaCR6Lp2tNfbYa7kr
t=2024-04-25T19:37:13+0000 lvl=info msg="Received signed execution payload from p2p" id=0x7c6af6b4ab46a7b0f5c799cce46a137b18e78f3ee4030498ae473b89c9f3711a:13642243 peer=16Uiu2HAm3VP6HuTR1VMhssD9geFfDiki2CVJGbKV3fPq2UNUL3F5
t=2024-04-25T19:37:13+0000 lvl=info msg="Optimistically queueing unsafe L2 execution payload" id=0x7c6af6b4ab46a7b0f5c799cce46a137b18e78f3ee4030498ae473b89c9f3711a:13642243

**Comment by jotto** [comment]
Note, I am using `OP_NODE_L1_TRUST_RPC=true` (because my L1 is Erigon). Maybe the snapshot is somehow incompatible with that setting?

**Comment by wbnns** [comment]
@jotto Heya! Hm, let's take a closer look at your config together. Can you please drop by the [Base Discord](https://base.org/discord) and open a #developer-support ticket? Please tag me (wbnns) when you do. 👍

**Pull request merged**